internal/controllers/cluster/cluster_controller_status.go

```diff
@@ -65,8 +65,38 @@ func (r *Reconciler) updateStatus(ctx context.Context, s *scope) {
 	setControlPlaneAvailableCondition(ctx, s.cluster, s.controlPlane, s.controlPlaneIsNotFound)
 	setControlPlaneInitializedCondition(ctx, s.cluster, s.controlPlane, s.descendants.controlPlaneMachines, s.infraClusterIsNotFound, s.getDescendantsSucceeded)
 	setWorkersAvailableCondition(ctx, s.cluster, expv1.MachinePoolList{}, s.descendants.machineDeployments, s.getDescendantsSucceeded)
-	setMachinesReadyCondition(ctx, s.cluster, allMachines, s.getDescendantsSucceeded)
-	setMachinesUpToDateCondition(ctx, s.cluster, allMachines, s.getDescendantsSucceeded)
+	machinesReadySetter{
```

Comment:
What do you think about having setControlPlane/WorkerMachinesReadyCondition and setControlPlane/WorkerMachinesUpToDateCondition functions with just the call to the Setter, this will help in keeping this updateStatus readable.

Also, it will make it easier to test those setter (currently we are duplicating setters configuration in test, which is sort of error prone)